library/webpack-loaders/react-universal-server-loader.js

```diff
@@ -25,9 +25,12 @@ function createServerCode({ importPath, scriptSrc, id }) {
        |\` }} />`.split(/^[ \t]*\|/m).join('')
 
   return `|import Component from './${importPath}'
+          |import hoistNonReactStatic from 'hoist-non-react-statics'
           |import { renderToString } from 'react-dom/server'
           |
-          |export default function WrapWithScript(props) {
+          |export default hoistNonReactStatic(WrapWithScript, Component)
+          |
+          |function WrapWithScript(props) {
```

Comment:
I would keep the `export default` here. The `hoistNonReactStatic` is side-effecting and it returning a result is suggesting otherwise.

Reply:
good suggestion!